### PR TITLE
Harden System.Net.Http.Functional.Tests.HttpClientHandlerTest_Http2.Dispose_ProcessingResponse_OK

### PR DESCRIPTION
**Pull request by krwq**
Fixes: https://github.com/dotnet/corefx/issues/38859

Per PR feedback, since those were verified test issues and we already have a test case testing this specific scenario in a slightly better way and also covers couple more cases I'm removing the test completely

OUTDATED:
This PR fixes two problems in this test:
- race between updating totalSent and Assert checking if totalSent matches received (note that last totalSent update is happening after the response is sent which means that client could have previously finish the loop before that value got updated which could have caused referenced issue) - this is likely fixing https://github.com/dotnet/corefx/issues/38859 but I was not able to get this specific issue repro locally
- When server sends a lot of data before client had a chance to to receive anything the client could have overfilled the receive window which could have caused random issue (I've removed the delay in the test now to make it more likely to introduce the first issue but it made this issue re-surface)

with those fixes I was able to run this test without any failures for around an hour already (previously it was failing relatively consistently with the second issue)

### CODE REVIEW

File: src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Http2.cs [hunk 1]
```diff
@@ -1526,6 +1528,9 @@ public async Task Dispose_ProcessingResponse_OK()
                     } while (readLength != 0);
                 }
 
+                // Wait until server finished processing
+                // so that totalSent is fully updated
+                serverProcessing.Wait();
```
Comment (davidsh) [inline src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Http2.cs]: We shouldn't be using synchronous `.Wait()` here. Instead, this should be `await serverProcessing;`

Comment (davidsh) [reply src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Http2.cs]: Oh...never mind. This is not a task. :)

Comment (krwq) [reply src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Http2.cs]: Note: this is not a Task, it's MRE. I'm getting compile errors with that:

```
HttpClientHandlerTest.Http2.cs(1533,17): error CS0012: The type 'IAsyncOperationWithProgress<,>' is defined in an assembly that is not referenced. You must add a reference to assembly 'Windows, Version=255.255.255.255, Culture=neutral,
PublicKeyToken=null, ContentType=WindowsRuntime'. [F:\src\corefx\src\System.Net.Http\tests\FunctionalTests\System.Net.Http.Functional.Tests.csproj]
```

Comment (krwq) [reply src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Http2.cs]: unless you mean await Task.Run(() => serverProcessing.Wait());

Comment (krwq) [reply src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Http2.cs]: added the timeout, let me know if you want me to change to Task.Run as well

File: src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Http2.cs [hunk 2]
```diff
@@ -1537,17 +1542,24 @@ public async Task Dispose_ProcessingResponse_OK()
 
                 // Start streaming response and wait for client to be disposed.
                 byte[] responseBody = new byte[100];
-                while (!diposeCalled)
+                while (!disposeCalled)
```
Comment (geoffkizer) [inline src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Http2.cs]: I don't really understand why we do this. Do you know? Seems like a weird thing for the test to do.

Comment (geoffkizer) [reply src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Http2.cs]: It seems like what this test should do is:

(1) issue a request and get back the response message, but don't read it yet
(2) dispose the HttpClient
(3) signal the server to send some data -- a known amount of data < 64K, so we don't need to worry about control flow
(4) retrieve the data on the client and validate it

Comment (krwq) [reply src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Http2.cs]: perhaps non-deterministic factor checks more scenarios 😄 Will update per recommendation

**Comment by geoffkizer** [comment]
In general I wonder why we don't just modify HttpClientHandlerTest.Dispose_DisposingHandlerCancelsActiveOperationsWithoutResponses to work against HTTP2, instead of having this HTTP2-specific test. That test already deals with the issues I raised above, and tests additional scenarios as well. Overall it's a much better test.

Is there some reason we need an HTTP2-specific test here?

**Comment by krwq** [comment]
@geoffkizer I didn't know about it. Considering I've already verified this is a test issue and there is a better test I'll port the other test to work with http2 and remove this one

**Comment by krwq** [comment]
ok, I've verified that existing tests already runs with http2 loopback server and covers the same scenario so removing this test

**Pull request merged**